src/components/action-button.js

```diff
@@ -1,11 +1,24 @@
+/* eslint-disable no-await-in-loop */
 import allActions from "../utils/actions";
 import navigationActions from "../utils/navigation-actions";
 import styleFormatter from "../utils/style-formatter";
 
-export const runActions = async (actionList) => {
-  for (let i = 0; i < actionList.length; i++) {
-    // eslint-disable-next-line no-await-in-loop
-    await actionList[i]();
+export const runActions = async ({ actionCallList, model, layout, app }) => {
+  if (layout.runtimeExpressionEvaluation) {
+    const properties = await model.getProperties();
+    for (let i = 0; i < actionCallList.length; i++) {
+      const overrideValue =
+        properties.actions[i].value?.qStringExpression?.qExpr &&
```

Comment:
I have no idea what `actionCallList` does but, do note that `overrideValue` can become any falsy value. For example if `qExpr === ''`, `overrideValue` becomes an empty string.

Same for `overrideVariable`.

Reply:
Is `actions` a required property on `properties`?

Reply:
And does it always match the length of `actionCallList`

Reply:
That is a couple of questions. In order:

1. ActionCallList is the list of actions that need to run. Actions from actions get pushed into this array if they are found. Good catch. I'll look at that
2. Yes it is. 
3. It does not have to match actually. Since we first try to find the actions from a list the actionCallList might be shorter than actions. This could happen if you create the button through the api. It also makes sure it is backwards compatible. Most important though is probably conversion from the old button. There are some actions that we do not support in the new if I remember correct

Reply:
Fixed based on suggestion from Max. Also made sure that the actionCallList and actions list have the same length